File: testprojects/csharp/benchmark/domain/Indirect/Intermediate/BackgroundService.cs
namespace Domain.Indirect.Intermediate
{
	public class BackgroundService
	{
        public static Domain.Indirect.IndirectTo.ServiceOne serviceOneStaticAttribute;

        public Domain.Indirect.IndirectTo.ServiceOne serviceOne;

		public ServiceTwo serviceTwo;

        public static Domain.Indirect.IndirectTo.ServiceOne GetServiceOneviaStaticAttribute()
		{
			return serviceOneStaticAttribute;
		}

        public virtual Domain.Indirect.IndirectTo.ServiceOne GetServiceOne()
		{
			return serviceOne;
		}

		public virtual ServiceTwo GetServiceTwo()
		{
			return serviceTwo;
		}

        public virtual string GetServiceOneNameByArgument(Domain.Indirect.IndirectTo.ServiceOne so)
		{
			return so.GetName();
		}

        public virtual string GetPOINameByArgument(Domain.Indirect.IndirectTo.POI poi)
		{
			return poi.GetName();
		}
	}
}

File: testprojects/java/accuracy/domain/indirect/intermediate/BackgroundService.java
package domain.indirect.intermediate;

import domain.indirect.indirectto.ServiceOne;
import domain.indirect.indirectto.POI;
import domain.indirect.intermediate.ServiceTwo;

public class BackgroundService {
	public static ServiceOne serviceOneStaticAttribute;
	public  ServiceOne serviceOne;
	public  ServiceTwo serviceTwo;

	public static ServiceOne getServiceOneviaStaticAttribute(){
		return serviceOneStaticAttribute;
	}

	public ServiceOne getServiceOne() {
		return serviceOne;
	}
	public ServiceTwo getServiceTwo() {
		return serviceTwo;
	}
	
	public String getServiceOneNameByArgument(ServiceOne so){
		return so.getName();
	}

	public String getPOINameByArgument(POI poi){
		return poi.getName();
	}

}